psa_builder.py

```diff
@@ -204,16 +204,35 @@ def check_and_clone_repo(name, deps, dir):
     if not isdir(join(dir, name)):
         logging.info("Cloning %s repo", name)
         cmd = ["git", "-C", dir, "clone", "-b", gitref, deps.get(name)[0]]
-        run_cmd_and_return(cmd)
-        logging.info("Cloned %s repo successfully", name)
+        ret = run_cmd_and_return(cmd)
+        if ret == 0:
+            logging.info("Cloned %s repo successfully", name)
+        else:
+            logging.critical("Failed to clone %s repo, error: %d", name, ret)
+            sys.exit(1)
     else:
         logging.info("%s repo exists, fetching latest...", name)
         cmd = ["git", "-C", join(dir, name), "fetch"]
-        run_cmd_and_return(cmd)
-        logging.info("%s repo exists, checking out %s...", name, gitref)
+        ret = run_cmd_and_return(cmd)
+        if ret != 0:
+            logging.critical(
+                "Failed to fetch the latest %s, error: %d", name, ret
+            )
+            sys.exit(1)
+        logging.info("Checking out %s...", gitref)
+        # try gitref as a remote branch
         head = "origin/" + gitref
         cmd = ["git", "-C", join(dir, name), "checkout", head]
-        run_cmd_and_return(cmd)
+        ret = run_cmd_and_return(cmd)
+        if ret != 0:
+            # gitref might be a tag or SHA1 which we checkout directly
+            cmd = ["git", "-C", join(dir, name), "checkout", gitref]
+            ret = run_cmd_and_return(cmd)
+            if ret != 0:
+                logging.critical(
+                    "Failed to checkout %s, error: %d", gitref, ret
+                )
+                sys.exit(1)
         logging.info("Checked out %s successfully", gitref)
```

Comment:
``` suggestion
        ret = run_cmd_and_return(cmd)
        if ret != 0:
            logging.critical(
                    "Failed to checkout %s, error: %d", head, ret
                )

            # gitref might be a tag or SHA1 which we checkout directly
            cmd = ["git", "-C", join(dir, name), "checkout", gitref]
            ret = run_cmd_and_return(cmd)
            if ret != 0:
                logging.critical(
                    "Failed to checkout %s, error: %d", gitref, ret
                )
                sys.exit(1)
            
            logging.info("Checked out %s successfully", gitref)
        else:
            logging.info("Checked out %s successfully", head)

Reply:
Not sure about logging an error at the first `if ret != 0` - users may be confused and think there's a real problem if they are not familiar. And in my personal opinion, printing `gitref` as the message is only for a rough reference, but I'm fine with `head` (i.e. `origin/<gitref>`) too.

Reply:
The "repo exists, checking out " above (even before this PR) only prints `gitref` without "origin/", so it's fine to do the same here.